src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala

```diff
@@ -62,6 +61,11 @@ trait GraphDAO {
     override def compute(a: A): B = f(a)
   }
 
+  def validateUserDefinedString(s: String) = {
+    // due to Orient's chained access "feature", we should avoid dots in certain user-defined strings
+    if (s.contains('.')) throw new IllegalArgumentException(s"User-defined $s should not contain dot characters")
+  }
```

Comment:
Won't this bubble up as an HTTP 500 response? I'm guessing it wants to be turned into a 4xx at some point, so the user knows it's their fault.